Makefile

```diff
@@ -31,6 +31,7 @@ lint-v110:
 vendor:
 	go mod vendor
 	modvendor -copy="**/*.c **/*.h" -v
+	go mod tidy
```

Comment:
I think we should do it **before** `go mod vendor` in this target. `go mod tidy` can change dependencies`.

Reply:
it will remove unused dependencies or previous versions from go.sum, or it can do something else? 
the thing is that i can update dependency by changing the version in go.mod and then run make vendor (without go get), and in this case, go mod tidy before make vendor will have no effect

so you think that the next two workflows will lead to different results?

1. 
- go get new version
- tidy
- vendor

2. 
- change version in go.mod
- vendor
- tidy

Reply:
I haven't checked that but I found:
> Run go mod tidy to possibly prune any extraneous requirements (as described here) and also ensure your current go.mod reflects all possible build tags/OS/architecture combinations (as described here).

> In contrast, other commands like go build and go test will not remove dependencies from go.mod that are no longer required and only update go.mod based on the current build invocation's tags/OS/architecture.

There is no such statement next to `go mod vendor` but there is:
> go mod vendor resets the main module's vendor directory to include all packages needed to build and test all of the module's packages based on the state of the go.mod files and Go source code.

So, in theory, if there are unused packages in go.mod, they can be still vendored.

> the thing is that i can update dependency by changing the version in go.mod and then run make vendor (without go get), and in this case, go mod tidy before make vendor will have no effect

I think it will. It will try to download the updated version, provided the module is used in the source code, otherwise it will remove it.

Reply:
> I think it will. It will try to download the updated version, provided the module is used in the source code, otherwise it will remove it.

oh, you are right. i thought it works on the vendor folder.
moving it up then